File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    const area = brown + yellow;
    let i = 0;
    count = 2;
    while (true) {
        count++;
        if (area % count !== 0) continue;

        i = area / count;
        if (count * 2 + (i - 2) * 2 === brown) {
            break;
        }
    }

    return [count, i].sort((a, b) => b - a);
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    half_perimeter_of_yellow = (brown - 4) // 2

    for width_or_height in range(half_perimeter_of_yellow):
        if width_or_height * (half_perimeter_of_yellow - width_or_height) == yellow:
            answer.append(width_or_height + 2)
            answer.append(half_perimeter_of_yellow - width_or_height + 2)
            break

    answer.sort(reverse=True)

    return answer
